internal/server/middleware/grpc/middleware.go

```diff
@@ -234,6 +235,26 @@ func CacheUnaryInterceptor(cache cache.Cacher, logger *zap.Logger) grpc.UnarySer
 	}
 }
 
+// AuditSinkUnaryInterceptor sends audit logs to configured sinks upon successful RPC requests for auditable events.
+func AuditSinkUnaryInterceptor(logger *zap.Logger, publisher *auditsink.Publisher) grpc.UnaryServerInterceptor {
```

Comment:
For testing purposes it might be nice for this to accept publisher as an interface so we can mock it